### PR TITLE
Better Streaming

### PR DESCRIPTION
**Pull request by benwilson512**
maybe this is a better way to go through the file. can't seem to get elasticsearch working though so who knows :)

### CODE REVIEW

File: lib/exalice/providers/elastic/importer.ex [hunk 1]
```diff
@@ -57,9 +64,9 @@ defmodule ExAlice.Geocoder.Providers.Elastic.Importer do
     Tirexs.ElasticSearch.put(index_name, JSX.encode!(index), settings)
   end
 
-  def read_file(file) do
+  def read(file) do
     File.stream!(file)
-    |> Enum.map(fn content -> String.split(content, "\n", trim: true) end)
+    |> Stream.map(&String.strip/1)
```
Comment (benwilson512) [inline lib/exalice/providers/elastic/importer.ex:69]: It seemed like this is all you're trying to do yes?

File: lib/tasks/bootstrap.ex [hunk 2]
```diff
@@ -9,6 +9,7 @@ defmodule Mix.Tasks.Exalice.Bootstrap do
   """
 
   def run(args) do
+    Mix.Task.run("app.start", [])
```
Comment (benwilson512) [inline lib/tasks/bootstrap.ex:12]: This is necessary to make sure your application's process tree is started.

**Comment by benwilson512** [comment]
In general you'll probably want a relatively low concurrency number, probably 2, 4, or 8. Chunk size is a balance of memory usage and speed. I'd just try orders of magnitude first and see how it goes. 10, 100, 1000, 10_000.

Elasticsearch logs a whole bunch of errors when I try to run anything so it's hard for me to say exactly how well this works.

**Comment by benwilson512** [comment]
If I comment out the part that pings elastic search it runs at about 160mb/sec which is pretty solid

**Comment by benwilson512** [comment]
got it closer to 190mb/sec with concurrency 8 and a chunk size of 5000. Obviously when you hook it up to elasticache it'll be a lot slower cause it'll have to wait on responses.

Memory stays at ~70mb the whole time though which is nice.

**Comment by kpanic** [comment]
Hey @benwilson512 !  this is awesome, I will try it and review it.
Amazing stuff, thank you!

**Comment by kpanic** [comment]
@benwilson512 I imported the whole germany in about 17 minutes, 9M of docs on my rusty spinning disk
sweet!
chunk_size: 5000, with "capacity" 8

I will try with capacity four for fun, I just have 4 kernel threads and I saw some improvement with capacity 4 with the streets of Berlin

**Comment by benwilson512** [comment]
Awesome! How did memory usage look? I know that was your original problem so I wrote it in such a way as to have hopefully constant memory use

**Comment by kpanic** [comment]
it was around 80M by looking at the observer / 80M-150M at system level fluctuating a little bit, but staying stable, which is really good

**Comment by kpanic** [comment]
@benwilson512 merged with some additional minor commits. Thank you again

**Pull request merged**